vowpalwabbit/options_boost_po.h

```diff
@@ -142,7 +153,15 @@ struct options_boost_po : public options_i
   template <typename T>
   void add_to_description(std::shared_ptr<typed_option<T>> opt, po::options_description& options_description);
 
+  void add_to_option_group_collection(const option_group_definition& group);
+
 private:
+  // Collection that tracks for now
+  // setup_function_id (str) -> list of option_group_definition
+  std::map<std::string, std::vector<option_group_definition>> m_option_group_dic;

```

Comment:
is there a reason we are using `map` instead of `unordered_map`? Do we want these to be ordered?

Reply:
no specific reason, but having them ordered has benefits when outputting them for now. i.e. showing up in almost alphabet order https://github.com/VowpalWabbit/vowpal_wabbit/wiki/VW-arguments-as-of-8.9.0-in-json-format